facebookresearch/fairo
# Build out chat interface to replace AgentThinking and respond to every command

# Description

Change the user-agent UI from using popups to using the chat interface.

## Type of change

Please check the options that are relevant.

- [X] Proposes a change (non-breaking change that isn't necessarily a bug)
- [ ] New feature (non-breaking change that adds a new functionality)
- [ ] Breaking change (fix or feature that would break some existing functionality downstream)
- [ ] This is a unit test
- [ ] Datasets Release
- [ ] Models Release

## Type of requested review

- [x] I want a thorough review of the implementation.
- [ ] I want a high level review. 
- [ ] I want a deep design review.

## Before and After

<img width="1173" alt="Screen Shot 2022-03-07 at 4 45 30 PM" src="https://user-images.githubusercontent.com/87049088/157143385-78ea3e09-c750-4a9e-ae38-f76a8f003f67.png">


# Testing

1. Run the craftassist server
2. Give a command, like "dig a hole".3. 
3. See communication from agent as chat.

# Checklist:

- [X] I have performed manual end-to-end testing of the feature in my environment.
- [ ] I have added Docstrings and comments to the code.
- [ ] I have made changes to existing documentation where needed.
- [ ] I have added tests that show that the PR is functional.
- [ ] I have added relevant collaborators to review the PR before merge.
- [ ] [Polymetis only] I ran on hardware (1) all scripts in `tests/scripts`, (2) asv benchmarks.
